run_tests.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Default values
+NETWORK="regtest"
+LOG_LEVEL="none"
+LONG_TESTS=0
+BACKTRACE=0
+
+# Parse command-line arguments
+while getopts "l:b:t:L" opt; do
+  case $opt in
+    l)
+    if [ "$OPTARG" == "debug" ] || [ "$OPTARG" == "info" ] || [ "$OPTARG" == "none" ]; then
+        LOG_LEVEL=$OPTARG
+    else
+        echo "Invalid log level option. Use 'info', 'debug', or 'none'."
+        exit 1
+    fi
+    ;;
+    b)
+      BACKTRACE=$OPTARG
+      ;;
+    t)
+      TEST_NAME="$OPTARG"
+      ;;
+    L)
+      LONG_TESTS=1
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+    :)
+      echo "Option -$OPTARG requires an argument." >&2
+      exit 1
+      ;;
+  esac
+done
+
+# Derive the NETWORK value from ~/.bitcoin/bitcoin.conf
+if grep -q '^regtest=1' ~/.bitcoin/bitcoin.conf; then
+    NETWORK="regtest"
+elif grep -q '^signet=1' ~/.bitcoin/bitcoin.conf; then
+    NETWORK="signet"
+else
+    echo "No valid network configuration found in bitcoin.conf."
+    exit 1
+fi
+
+# Create a wallet if it doesn't exist
+bitcoin-cli createwallet test 2>/dev/null
+
+# Load the wallet
+bitcoin-cli loadwallet test 2>/dev/null
+
+# Function to get address by label
+get_address_by_label() {
+    local network=$1
+    LABEL_ADDRESS=$(bitcoin-cli -$network getaddressesbylabel "test" | jq -r 'keys[0]')
+    if [ -z "$LABEL_ADDRESS" ]; then
+        # Create a new address and label it
+        NEW_ADDRESS=$(bitcoin-cli -$network getnewaddress)
+        bitcoin-cli -$network setlabel "$NEW_ADDRESS" "test"
+        LABEL_ADDRESS=$NEW_ADDRESS
+    fi
+    echo $LABEL_ADDRESS
+}
+
+
+# Conditional block generation based on the network
+if [ "$NETWORK" == "regtest" ]; then
+    bitcoin-cli -regtest -generate 150
```

Comment:
Idk if this is usefull. It is for sure when you launch a node but if the user already has a daemon then maybe not

Reply:
removing this line: `bitcoin-cli -regtest -generate 150`